tests/specs/lemmas.k

```diff
@@ -330,6 +341,44 @@ module LEMMAS-HASKELL [symbolic, kore]
     rule BA:ByteArray   ==K #buf(32, DATA) => #buf(32, DATA) ==K              BA                                       [simplification, concrete(BA)]
     rule #buf(32, DATA) ==K BA:ByteArray   =>          DATA  ==Int #asInteger(BA) requires #sizeByteArray(BA) <=Int 32 [simplification, concrete(BA)]
 
+    rule #range( M [ N1 := BUF1:ByteArray ] [ N2 := BUF2:ByteArray ] [ N3 := BUF3:ByteArray ], L, K ) =>
+         #range( M                          [ N2 := BUF2           ] [ N3 := BUF3           ], L, K )
+      requires N1 +Int #sizeByteArray(BUF1) <=Int L [simplification]
+
+    rule #range ( M [ N := BUF:ByteArray ], N, K ) => #range( #range( M, N, K ) [ 0 := BUF ], 0, K ) 
+      requires N >Int 0 [simplification]
+
+    rule #range ( M, N, K ) => #range( #drop(N, M), 0, K ) 
+      requires N >Int 0 [simplification]
```

Comment:
`drop` is another one we should never write about directly in lemmas, it's implementation specific (you can tell because it has two different implementations). On top of that, there are no safeguards on the `drop` rules to prevent them form evaluation on symbolic arguments (no `concrete` attribute, and no rule name for it listed in `tests/specs/concrete-rules.txt`).

Instead, use `_[_.._]`.

Reply:
In the latest commit I've replaced it with the rule
```
    rule #range (M, N1 +Int N2, K ) => #range( #range( M, N2, #sizeByteArray(M) -Int N2 ), N1, K ) 
      [simplification, concrete(N2), concrete(M)]
```
What I was trying to do was remove concrete bytes from inside a range query that we should know cannot be inside the range.

Reply:
Makes sense, but I think your added lemma isn't quite correct.